src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,25 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_properties_from_model_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
```

Comment:
Better to update the comments. The document in-place still confuse. As the document here means the internal implementation, but the method doc should only talks about the method input and output.  Otherwise, it leads to consider it modify the input/output model in place.